src/time/Waveform.hpp

```diff
@@ -0,0 +1,122 @@
+#pragma once
+
+#include <Eigen/Core>
+#include "logging/Logger.hpp"
+
+namespace precice {
+
+namespace testing {
+// Forward declaration to friend the boost test struct
+class WaveformFixture;
+} // namespace testing
+
+namespace time {
+
+/**
+ * @brief Stores data samples in time and allows to perform interpolation to obtain new values.
+ */
+class Waveform {
+  friend class testing::WaveformFixture; // Make the fixture friend of this class
+public:
+  /**
+   * @brief Waveform object which stores values of current and past time windows for performing interpolation. 
+   *
+   * Storage still needs to be initialized with Waveform::initialize, before the Waveform can be used.
+   *
+   * @param interpolationOrder Defines the maximum interpolation order supported by this Waveform and reserves storage correspondingly
+   */
+  Waveform(const int interpolationOrder);
+
+  /**
+   * @brief Get the _interpolationOrder.
+   * 
+   * @return int _interpolationOrder
+   */
+  int getInterpolationOrder() const;
+
+  /**
+   * @brief Used to initialize _timeWindowsStorage according to required size.
+   * @param valuesSize Defines how many values one sample in time consists of
+   */
+  void initialize(const int valuesSize);
+
+  /**
+   * @brief Updates first entry in _timeWindows with given values.
+   * @param values Sample for this time window
+   */
+  void storeAtFirstSample(const Eigen::VectorXd &values);
+
+  /**
+   * @brief Updates all entries in _timeWindowsStorage with given values. Often used to initialize a new Waveform with given data.
+   * @param values Sample used for all time window
+   */
+  void storeAtAllSamples(const Eigen::VectorXd values);
+
+  /**
+   * @brief Shifts all entries in _timeWindows. The new entry is initialized as the value from the last window (= constant extrapolation). Usually called, when moving to the next time window.
+   */
+  void moveToNextWindow();
+
+  /**
+   * @brief Evaluate Waveform at specific point in time. Uses interpolation with Waveform's interpolation order, if necessary.
```

Comment:
```suggestion
   * @brief Evaluate waveform at specific point in time. Uses interpolation if necessary.
```
Waveform = name of the class (good practice would be to never directly use as a subject, cf. [google style guide](https://developers.google.com/style/code-in-text#grammatical-treatment-of-code-elements)) 
waveform = mathematical concept